src/allmydata/introducer/client.py

```diff
@@ -111,9 +111,29 @@ def startService(self):
         def connect_failed(failure):
             self.log("Initial Introducer connection failed: perhaps it's down",
                      level=log.WEIRD, failure=failure, umid="c5MqUQ")
+            self._load_announcements()
         d = self._tub.getReference(self.introducer_furl)
         d.addErrback(connect_failed)
 
+    def _load_announcements(self):
+        if self._cache_filepath.exists():
+            with self._cache_filepath.open() as f:
+                servers = yaml.load(f)
+                f.close()
+            if not isinstance(servers, list):
+                msg = "Invalid cached storage server announcements. No list encountered."
+                self.log(msg,
+                         level=log.WEIRD)
+                raise storage_client.UnknownServerTypeError(msg)
```

Comment:
I think we're missing an `import storage_client` somewhere.. pyflakes rejects this. But really, we should probably define a local exception type, rather than creating a dependency between the introducer client and the storage client.